### DISCUSSION TITLE
Organize the tuxemon a bit more?

### DISCUSSION THREAD

**Discussion by HippasusTwo** [Tuxemon/Tuxemon#2264] [Ideas]
I found myself a little frustrated using the wiki when I found myself trying to look up if it already had certain critters, like amoebas (I managed to find Legoplaste who sorta looks like one, and now it's tagged Amoeba) or ducks. I think it would really benefit from having a few more categories/tags to place Tuxemon in, for instance 'real world analogue(s)': 'Animal/nonanimal life/chemistry/object' (subcategories of first?), 'prehistoric/modern/futuristic', and 'theme' and 'continent/region'. The Egypt themed critters could all get 'Africa' tagged on them, for instance, and then there could be an 'Africa' page that would load all of them. Just adding 'region' and 'real life analogue' would probably help a lot.
This would make it easier to see if, say, Australia had only one critter also. I think I saw one marsupial.
Critters I like but am unsure have any in the game:
Nematode
An archaean or some kind of universal common ancestor
Trilobite
Air breathing fish (I always thought they'd make a natural choice.)
Frogs
Parasitic wasp

I'm seeing a lot of cats, dogs, birds and surprisingly snakes and insects. But if I go 'I think I want a kangaroo', I have to hope there's a critter with the species name 'kangaroo' and use ctrl+f on a list page of species names, because it won't show up via the wiki search bar. Maybe that requires making a category page named 'kangaroo'?

Obviously this would be a fair bit of annoying effort and I doubt there's going to be a lot of volunteering, so I guess mostly I'm just asking if there are any objections if I ever get around to actually joining the wiki to me re-organizing a bit. Although if someone spontaneously decide to do all of this on their own, that would quite nice too.

**Comment by Sanglorian** [comment]
Hi @HippasusTwo , I'm certainly open to you adding more detail to the wiki. One option would be to put this stuff in the "Design origin" section - that tends to have been where in the past we've noted stuff like what animal a creature is based on. Then we could make a page that displays a table of every monster and its design origins. But very happy for you to add categories too if that helps you sort things.

**Reply by HippasusTwo** [reply]
A table that displays design origin could work and would probably be less effort.

**Reply by Sanglorian** [reply]
Do you have wiki edit rights? If not, I can set you up with those. I can also make a page that shows monster design origins so far. I know you've had lots of insight into what animals the mons are based on.

**Reply by HippasusTwo** [reply]
No, I do not. I remember I went to go look into making an account and the wiki seemed to have that disabled? Or I missed where the account make page was.

I think I was supposed to get a discord account and go on the tuxemon server, but I never got around to it (because I am not very familiar with discord). I'll do that now.
I can usually identify what an animal is even if it's a cartoon, though admittedly sometimes there are ambiguities. Cartoon cheetahs often aren't shaped much like real cheetahs, for instance.

**Reply by Sanglorian** [reply]
If you shoot me a message on Discord (same username as here) I can get you set up on the wiki. Sorry it's taken so long!

**Comment by JaskRendix** [comment]
From what I've observed , I think the species could be more interesting if they has some practical uses, especially in battle. Right now, they're all over the place (50+) and don't offer any real advantages or disadvantages. What if we tied their abilities to their body shape, creating a sub-species with unique bonuses and penalties (melee, etc)? Here are some ideas:

**Blob**: Amorphous, gelatinous Tuxemon with no discernible limbs or features. They can range in size from tiny to colossal.
Subspecies: Acidic Blob, Electric Blob, Mimic Blob, Phase-Shifting Blob, Toxic Blob

**Brute**: Massive, muscular Tuxemon with brute strength as their primary asset. Often quadrupedal or bipedal with thick hides and powerful jaws.
Subspecies: Armored Brute, Fighting Brute, Cave Brute, Goliath Brute, Troll Brute

**Dragon**: Reptilian Tuxemon with wings, often possessing fire-breathing abilities. They vary widely in size and appearance.
Subspecies: Ice Dragon, Lightning Dragon, Shadow Dragon, Storm Dragon, Wyvern

**Flier**: Tuxemon primarily adapted for flight, with large wings and lightweight bodies. They can vary greatly in size and appearance.
Subspecies: Bat Flier, Harpy Flier, Insectoid Flier, Pterodactyl Flier, Sky Serpent

**Grub**: Worm-like Tuxemon, often subterranean, with elongated bodies and multiple limbs. They can range from small and innocuous to large and predatory.
Subspecies: Burrower Grub, Carnivorous Grub, Glowworm Grub, Maggot Grub, Tunnelworm Grub

**Humanoid**: Tuxemon with a human-like form, but often with exaggerated features or additional limbs. They can vary in intelligence and aggression.
Subspecies: Fighting Humanoid, Goblin Humanoid, Ogre Humanoid, Undead Humanoid, Zombie Humanoid

**Hunter**: Tuxemon specifically adapted for hunting, with keen senses, sharp claws, and stealthy movement. They can vary in size and hunting methods.
Subspecies: Ambush Hunter, Pack Hunter, Stalker Hunter, Tracker Hunter, Venom Hunter

**Landrace**: Terrestrial Tuxemon with four limbs and a sturdy build, adapted to various environments. They can range in size and diet.
Subspecies: Boar Landrace, Lizard Landrace, Mammalian Landrace, Reptile Landrace, Turtle Landrace

**Leviathan**: Enormous, aquatic or semi-aquatic Tuxemon, often with powerful jaws and thick hides. They are apex predators in their environments.
Subspecies: Deep Sea Leviathan, Kraken Leviathan, Megalodon Leviathan, Sea Serpent Leviathan, Whale Leviathan

**Piscine**: Fish-like Tuxemon, adapted to aquatic life with fins and gills. They can vary in size and shape.
Subspecies: Electric Eel Piscine, Piranha Piscine, Shark Piscine, Squid Piscine, Whale Shark Piscine

**Polliwog**: Tadpole-like Tuxemon, often with large heads and tails. They can be terrestrial or aquatic.
Subspecies: Amphibian Polliwog, Giant Polliwog, Insectoid Polliwog, Jellyfish Polliwog, Toad Polliwog

**Serpent**: Long, limbless Tuxemon with elongated bodies and no legs. They can be terrestrial, aquatic, or aerial.
Subspecies: Constricting Serpent, Desert Serpent, Flying Serpent, Sea Serpent, Venomous Serpent

**Sprite**: Small, often ethereal Tuxemon with delicate features and magical abilities. They can be benevolent or mischievous.
Subspecies: Elemental Sprite, Fairy Sprite, Pixie Sprite, Salamander Sprite, Undine Sprite

**Varmint**: Small, rodent-like Tuxemon, typically pests or scavengers. They can be terrestrial or subterranean.
Subspecies: Burrow Varmint, Flying Varmint, Insectivorous Varmint, Scavenger Varmint, Water Varmint

**Reply by Sanglorian** [reply]
Thanks @JaskRendix .

I like the idea of more customisation of the monsters, but in that case I think we should go straight to the monsters themselves - so e.g. Bolt and Bumbulus and Cairfrey could all have different stat modifiers.

**Reply by HippasusTwo** [reply]
It could make sense for members of the same 'species family' to get a similar set of modifiers, but on the other hand for creatures that undergo dramatic metamorphosis this doesn't make much sense.

The downside of customizing every single monster, rather than expanding the kinds of body shapes / categories that have stat modifiers, is that it can make it harder to re-balance in one go. 
Another use of categories is, actually, some sort of skill learner system. It might make sense for all critters with a 'Legless' modifier added to them to be able to learn a 'Wiggle' or 'Low to earth' move from a tutor, for instance, but not all legless creatures have the same body shape: some might be more dragon-y, some might be insectoid or fish or blob. Or we might give a speed modifier to limbed creatures.

**Reply by Sanglorian** [reply]
> The downside of customizing every single monster, rather than expanding the kinds of body shapes / categories that have stat modifiers, is that it can make it harder to re-balance in one go. Another use of categories is, actually, some sort of skill learner system. It might make sense for all critters with a 'Legless' modifier added to them to be able to learn a 'Wiggle' or 'Low to earth' move from a tutor, for instance, but not all legless creatures have the same body shape: some might be more dragon-y, some might be insectoid or fish or blob. Or we might give a speed modifier to limbed creatures.

I'm glad you mentioned this! I've been thinking about a possible "tag" system for monsters that could determine which techniques they are eligible to learn from items (for techniques learned by leveling, we wouldn't change anything). So we might tag a monster as #claws #gemstones #teeth #light to indicate a monster that can learn Fiery Bite, Beam of Light and Ruby, for example.

**Reply by HippasusTwo** [reply]
Yeah, a tag system could make sense for items/tutors. Having a select few tags that are 'opposite' of each other, like 'claws' vs. 'legless', would make it possible to maybe attach some minor stat changes as well without it getting too imbalanced (since opposing tags don't stack), although a tag being associated with a really powerful move could make up for it not giving a stat boost as well... Or oppositely, a really good stat tag might have downsides in the variety of move department, like 'Brainless' giving extra HP but not offering up any moves. Something to think about.

**Reply by JaskRendix** [reply]
@Sanglorian that's an interesting idea! I like the concept of using tags to determine which techniques a monster can learn from items. When it comes to defining these tags, do you think they should be based solely on the monster's physical characteristics (e.g. #claws, #teeth, #quadruped) or should they also include habitat or environmental factors (e.g. #desert, #forest, #aquatic)?

I'm also wondering if these tags might end up duplicating some of the existing type or body shape categories. For instance, if we have a #flier tag, would that be redundant with the existing Flier body shape? How do you envision the relationship between these tags and the existing type and body shape systems?

Speaking of habitat and environmental factors, I was thinking about the existing Terrains category on the wiki (https://wiki.tuxemon.org/Category:Terrains). Would we need to convert terrains into tags as part of this new system? So we can avoid duplicates of data.

**Reply by HippasusTwo** [reply]
Only de-duplicate data if you're prepared to go in and have things like a terrain var re-generated in the code so that we don't have to loop through all the tags every time we want to get the terrain (or type, or other feature shoved into a tag). At this point in time disc space is so incredibly cheap and plentiful that I'm more worried about eating into people's free time, which having to make more code to perform steps to get info that used to be handled by an automatically grabbed variable would do. Making a loop may be easy but if I have to do a match against every single tag versus every single bit of terrain the code for something that used to be very simple to get could be very bloated.

Now if terrain is only ever used for checking TM availability, then it doesn't matter, but it could be used for things like figuring out if a given creature belongs in a terrain so, say, a future randomizer could give you a completely random set of appropriate tuxemon on every playthrough which could be a pretty cool menu option for replayability.

**Comment by HippasusTwo** [comment]
@JaskRendix @Sanglorian 

How does breeding work / going to work?

In real life, species is supposed to indicate relation. So it could be used for breeding groups / sorting critters that are somewhat alike, provided the species grouping is very wide (so more like a clade). It would be kind of odd if you could breed a horse to a mosquito.

Right now, it's kind of weird that Lapinou and Squabbit, which are wild and domesticated versions of the same critter, don't share a species. That makes no sense. Wolves and dogs after domestication are still the same species. Ditto horses and wild horses.
And it is completely random whether an evo will share the same species or not as its original form, even if it is clearly meant to be the same variety of critter (a metamorphosis) as opposed to, say, a fish evolving to an amphibian (not that we have any such evolutionary chain, but hypothetically). So species isn't even useful for sorting right now.

I think for instance all rabbits and hares should be grouped under 'Rabbit' or, if we have any pikas, 'Lagomorph' or 'Rabbitkin'. This would also get back to my original post, which is that it is hard to find if a tuxemon version of a certain critter exists. If all the Koalas shared a 'Koala' species, or 'Marsupial' because we probably won't have that many marsupials, it would make searching so much easier.

**Reply by Sanglorian** [reply]
I was thinking we could do breeding by body shape, but to be honest I haven't given it much thought. I don't think we need to worry about realism too much as these are magical creatures.

I see "Species" descriptions as just a bit of flavour or a nickname, rather than a description of the biological nature of the creature. But we could definitely add a different field to use to identify all the rabbits/lagomorphs, marsupials, etc. Reminds me of the phylogenetic trees people make for Pokemon (for example [this one](https://www.reddit.com/r/pokemon/comments/11pl2gh/pok%C3%A9mon_phylogenetic_tree_of_life_updated_for/)), which are a lot of fun!

**Reply by JaskRendix** [reply]
> Reminds me of the phylogenetic trees people make for Pokemon (for example [this one](https://www.reddit.com/r/pokemon/comments/11pl2gh/pok%C3%A9mon_phylogenetic_tree_of_life_updated_for/))

For doing that we need some data points, such as:
* body shape
* size (height and weight)
* number of limbs: two, four or none (*)
* number of tails: zero, one or multiple (*)
* number of heads: one or multiple (*)
* presence of wings: yes or no (*)
* presence of horns, spikes or other protusions: yes or no (*)
* coloration: dominant colors (*)
* skin or scale texture: smooth, rough, scaly or other (*)

(*) need to be collected

@Sanglorian some of these can be implemented as tags, like wings, limbs, tail, etc.

then we can consider the type and the secondary type, moveset, evolutionary relationship and habitat, then we can generate everything with matplotlib

**Reply by HippasusTwo** [reply]
Ah, you're right, we could do breeding by body shape. Would be less effort.

Phylogeny can be fun, indeed. Thanks for the link.
